packages/story-editor/src/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,199 @@
+      } else {
+        setErrorMsg(null);
+      }
+      return {
+        ext,
+        type: ['m4v', 'mp4'].includes(ext) ? 'video' : 'image',
+      };
+    },
+    [link, allowedFileTypes, error]
+  );
+
+  const onInsert = useCallback(async () => {
+    if (errorMsg?.length) {
+      return;
+    }
+    const insertionError = __(
+      'Media failed to load. Please ensure the link is valid and the site allows linking from external sites',
+      'web-stories'
+    );
+    if (!isValidUrl(link)) {
+      setErrorMsg(insertionError);
+      return;
+    }
+    try {
+      const { type } = getFileInfo();
+      const isVideo = type === 'video';
+      const getMediaDimensions = isVideo
+        ? getVideoDimensions
+        : getImageDimensions;
+      const { width, height } = await getMediaDimensions(link);
+
+      // Add necessary data for video.
+      let posterData;
+      const videoData = {};
+      const originalFileName = getFileNameFromUrl(link);
+      if (isVideo) {
+        // Create poster if possible.
+        if (hasUploadMediaAction) {
+          const fileName = getPosterName(originalFileName);
+          const posterFile = await getFirstFrameOfVideo(link);
+          posterData = await uploadVideoPoster(0, fileName, posterFile);
+          videoData.poster = posterData.poster;
+          videoData.posterId = posterData.posterId;
+        }
+        const hasAudio = await hasVideoGotAudio(link);
+        videoData.isMuted = !hasAudio;
+        const { length, formattedLength } = getVideoLength(link);
+        videoData.length = length;
+        videoData.formattedLength = formattedLength;
+      }
+      // @todo Create getResourceFromUrl util instead.
+      insertElement(type, {
+        resource: createResource({
+          alt: originalFileName,
+          type,
+          width,
+          height,
+          src: link,
+          local: false,
+          ...videoData,
+        }),
+      });
+      setErrorMsg(null);
+      setLink('');
+      onClose();
+    } catch (e) {
+      setErrorMsg(insertionError);
+    }
+  }, [
+    insertElement,
+    link,
+    errorMsg,
+    getFileInfo,
+    onClose,
+    uploadVideoPoster,
+    hasUploadMediaAction,
+  ]);
+
+  return (
+    <Dialog
+      onClose={() => {
+        onClose();
+        setLink('');
+        setErrorMsg(false);
+      }}
+      isOpen={isOpen}
+      title={__('Insert external image or video', 'web-stories')}
+      onPrimary={() => onInsert()}
+      primaryText={__('Insert', 'web-stories')}
+      secondaryText={__('Cancel', 'web-stories')}
+    >
+      <InputWrapper>
+        <Input
+          onChange={({ target: { value } }) => {
+            setLink(value);
+            getFileInfo(value);
+          }}
+          value={link}
+          hint={errorMsg?.length ? errorMsg : description}
+          hasError={Boolean(errorMsg?.length)}
+          onBlur={() => setLink(withProtocol(link))}
+          label={__('URL', 'web-stories')}
+        />
```

Comment:
Please focus this input on mount. Clicking the hotlink button and pasting a URL actually just inserts the URL as a text element on the canvas because focus is not in the dialog.

Reply:
For a11y reasons, it probably makes sense to also mark this input as `required`.

Reply:
Oh, and [`type="url"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url) would probably also help a bit!